src/components/application_manager/src/commands/mobile/register_app_interface_request.cc

```diff
@@ -501,6 +509,241 @@ void FillUIRelatedFields(smart_objects::SmartObject& response_params,
+      result_code = mobile_apis::Result::RESUME_FAILED;
+      add_info = "Hash from RAI does not match to saved resume data.";
+      need_restore_vr = false;
+    } else if (!resumer.CheckPersistenceFilesForResumption(application)) {
+      LOG4CXX_WARN(logger_, "Persistent data is missing.");
+      result_code = mobile_apis::Result::RESUME_FAILED;
+      add_info = "Persistent data is missing.";
+      need_restore_vr = false;
+    } else {
+      add_info = "Resume succeeded.";
+    }
+  }
+  if ((mobile_apis::Result::SUCCESS == result_code) &&
+      (mobile_apis::Result::INVALID_ENUM != result_checking_app_hmi_type_)) {
+    add_info += response_info_;
+    result_code = result_checking_app_hmi_type_;
+  }
+
+  // in case application exist in resumption we need to send resumeVrgrammars
+  if (false == resumption) {
+    resumption = resumer.IsApplicationSaved(application->policy_app_id(),
+                                            application->mac_address());
+  }
+
+  AppHmiTypes hmi_types;
+  if ((*message_)[strings::msg_params].keyExists(strings::app_hmi_type)) {
+    smart_objects::SmartArray* hmi_types_ptr =
+        (*message_)[strings::msg_params][strings::app_hmi_type].asArray();
+    DCHECK_OR_RETURN_VOID(hmi_types_ptr);
+    SmartArrayValueExtractor extractor;
+    if (hmi_types_ptr && 0 < hmi_types_ptr->size()) {
+      std::transform(hmi_types_ptr->begin(),
+                     hmi_types_ptr->end(),
+                     std::back_inserter(hmi_types),
+                     extractor);
+    }
+  }
+  policy::StatusNotifier notify_upd_manager = GetPolicyHandler().AddApplication(
+      application->policy_app_id(), hmi_types);
+  SendResponse(true, result_code, add_info.c_str(), &response_params);
+  SendOnAppRegisteredNotificationToHMI(
+      *(application.get()), resumption, need_restore_vr);
+#ifdef SDL_REMOTE_CONTROL
+  if (msg_params.keyExists(strings::app_hmi_type)) {
+    GetPolicyHandler().SetDefaultHmiTypes(application->policy_app_id(),
+                                          &(msg_params[strings::app_hmi_type]));
+  }
+#endif  // SDL_REMOTE_CONTROL
+
+  // Default HMI level should be set before any permissions validation, since it
+  // relies on HMI level.
+  application_manager_.OnApplicationRegistered(application);
+  (*notify_upd_manager)();
+
+  // Start PTU after successfull registration
+  // Sends OnPermissionChange notification to mobile right after RAI response
+  // and HMI level set-up
+  GetPolicyHandler().OnAppRegisteredOnMobile(application->policy_app_id());
+
+  if (result_code != mobile_apis::Result::RESUME_FAILED) {
+    resumer.StartResumption(application, hash_id);
+  } else {
+    resumer.StartResumptionOnlyHMILevel(application);
+  }
+
+  // By default app subscribed to CUSTOM_BUTTON
+  SendSubscribeCustomButtonNotification();
+  SendChangeRegistrationOnHMI(application);
+}
+
+// DEPRECATED
 void RegisterAppInterfaceRequest::SendRegisterAppInterfaceResponseToMobile() {
```

Comment:
Can you make this a wrapper for the new function signature?

Reply:
@jacobkeeler Unfortunately in this case we can not as new `SendRegisterAppInterfaceResponseToMobile` is used everywhere now and with different parameters passed.

Reply:
Same misunderstanding as before, I am only asking you to change the body of this function, the new version of `SendRegisterAppInterfaceResponseToMobile` should be the only one actually used in the project.

Example:
```
void RegisterAppInterfaceRequest::SendRegisterAppInterfaceResponseToMobile() {
  SendRegisterAppInterfaceResponseToMobile(ApplicationType::kNewApplication);
}
```

This would reduce 230 lines of deprecated code to 3

Reply:
@jacobkeeler Fixed in b969449ecb89eb95e288d9d0024d1b1d2d1f66f0

Reply:
@AByzhynar The reason that we don't remove these functions right now is because that would be a breaking change, which would require a major version change. The goes for any functionality with public scope in the project. These methods will eventually be removed in preparation for the 5.0.0 release.

The reason that these breaking changes matter is that OEMs can build upon this software with their own additions (primarily transport adapters, but there could be other additions made as well). With this setup OEMs can update to the latest compatible version (`4.x.x` in this case), and none of their additions will be broken.